ballerina/tests/12-custom-type-varray-test.bal

```diff
@@ -113,30 +113,247 @@ isolated function insertVarrayWithEmptyArray() returns sql:Error? {
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :string[] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_BYTEARR) values(${arrayOfByteArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :byte[][] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_BYTEARR) values(${byteArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:DatabaseError {
+       test:assertTrue(result.message().includes("Error while executing SQL query: insert into " +
+       "TestVarrayTable(COL_BYTEARR) values( ? ). ORA-00600: internal error code"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_INTARR) values(${intArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :int[] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_BOOLARR) values(${boolArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :boolean[] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_FLOATARR) values(${floatArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :float[] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_DECIMALARR) values(${decimalArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :decimal[] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+}
+
+type ArrayRecordType2 record {
+    int pk;
+    string?[] col_chararr;
+    byte[]?[] col_bytearr;
+    int?[] col_intarr;
+    boolean?[] col_boolarr;
+    float?[] col_floatarr;
+    decimal?[] col_decimalarr;
+};
+
 @test:Config {
     groups:["custom-varray"],
-    dependsOn: [insertVarrayWithEmptyArray]
+    dependsOn: [insertVarrayWithNullElements]
+}
+isolated function selectVarrayHavingNullElementsWithRecordType() returns error? {
+    ArrayRecordType2 arrayRecordInstance = {
+        pk : 4,
+        col_chararr : [null, null],
```

Comment:
Don't use `null`

Reply:
correct all places

Reply:
corrected in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/28/commits/dfabbaa76238f571c2ef9a58d207342c9b06bdef